types_registry.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2020 Sven Rebhan.
+// Licensed under the MIT license which can be found in the LICENSE file.
+
+package dpt
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+type Registry struct {
+	lut map[string]DatapointValue
+}
+
+// Fill the LUT with the given datatype
+func (r *Registry) add(d DatapointValue) error {
+	if r.lut == nil {
+		return fmt.Errorf("registry is not initialized!")
+	}
+
+	// Determine the name of the datatype
+	d_type := reflect.TypeOf(d)
+	name := d_type.Name()
+	if d_type.Kind() == reflect.Ptr {
+		name = d_type.Elem().Name()
+	}
+
+	// Make sure we only handle DPT types^
+	if !strings.HasPrefix(name, "DPT_") {
+		return fmt.Errorf("invalid type \"%v\" for registry!", name)
+	}
+
+	// Convert the name into KNX yy.xxx (e.g. DPT_1001 --> 1.001) format
+	name = strings.TrimPrefix(name, "DPT_")
+	name = name[:len(name)-3] + "." + name[len(name)-3:]
+
+	// Register the type
+	r.lut[name] = d
+
+	return nil
+}
+
+// Init function used to add all types
+func (r *Registry) Init() error {
```

Comment:
This doesn't need to be a method. Just make it a `func NewRegistry() (*Registry, error)`

Reply:
Addressed in the update.